frontend/packages/dev-console/src/utils/pipeline-utils.ts

```diff
@@ -311,3 +320,23 @@ export const pipelineRunDuration = (run: PipelineRun): string => {
     : new Date().getTime() - start;
   return formatDuration(duration);
 };
+
+export const updateServiceAccount = (
+  secret,
+  serviceAccount: ServiceAccountKind,
+): Promise<K8sResourceKind> => {
+  serviceAccount.secrets = [...serviceAccount.secrets, secret];
```

Comment:
@vikram-raj Still modifying the passed variable. Please don't do this, it's bad practice.